labels_command.php

```diff
@@ -0,0 +1,140 @@
+    'rfc/draft' => 'dddddd',
+    'type/api-break' => '1d76db',
+    'type/bug' => 'd93f0b',
+    'type/docs' => '02d7e1',
+    'type/enhancement' => '0e8a16',
+    'type/userhelp' => 'c5def5',
+    'type/UX' => '006b75',
+    'type/other' => '975515',
+];
+
+// Rename existing labels 'from' => 'to'
+const LABELS_RENAME = [
+    'effort/easy' => 'complexity/low',
+    'effort/medium' => 'complexity/medium',
+    'effort/hard' => 'complexity/high',
+    'change/major' => 'type/api-break',
+    'type/api-change' => 'type/api-break',
+];
+
+$labelsCommand = function(InputInterface $input, OutputInterface $output): int {
+    // This is the code that is executed when running the 'labels' command
+
+    // variables
+    global $OUT, $REPOS_WITH_LABELS_UPDATED;
+    $OUT = $output;
+
+    // validate system is ready
+    validate_system();
+
+    // setup directories
+    setup_directories($input);
+
+    // CMS major version to use
+    $cmsMajor = $input->getOption('cms-major') ?: CURRENT_CMS_MAJOR;
+
+    // modules
+    $modules = filtered_modules($cmsMajor, $input);
+
+    // update labels
+    foreach ($modules as $module) {
+        $account = $module['account'];
+        $repo = $module['repo'];
+
+        // Fetch labels
+        $labels = github_api("https://api.github.com/repos/$account/$repo/labels");
+
+        foreach ($labels as $key => $label) {
+            // $label is an array, for example:
+            // 'id' => 427423377
+            // 'node_id' => MDU6TGFiZWw0Mjc0MjMzNzc="
+            // 'url' => "https://api.github.com/repos/silverstripe/silverstripe-config/labels/affects/v4"
+            // 'name' => "affects/v4"
+            // 'color' => "5319e7"
+            // 'default' => false
+            // 'description' => NULL
+            $url = $label['url'];
+            $name = $label['name']; // e.g. 'affects/v4'
+
+            // Rename label
+            // https://docs.github.com/en/rest/issues/labels#update-a-label
+            if (array_key_exists($name, LABELS_RENAME)) {
+                $newName = LABELS_RENAME[$name];
+                info("Updating label $name to $newName in $repo");
+                if ($input->getOption('dry-run')) {
+                    info('Not updating label on GitHub because --dry-run option is set');
+                } else {
+                    github_api($url, ['new_name' => $newName], 'PATCH');
+                }
+                $name = $newName;
+                $labels[$key]['name'] = $newName;
+            }
+
+            // Delete label
+            // https://docs.github.com/en/rest/issues/labels#delete-a-label
+            if (!array_key_exists($name, LABELS_COLORS)) {
+                info("Deleting label $name from $repo");
+                if ($input->getOption('dry-run')) {
+                    info('Not deleting label on GitHub because --dry-run option is set');
+                } else {
+                    github_api($url, [], 'DELETE');
+                }
```

Comment:
If you want to, it might make sense to have a function to handle these conditions which can take a string and a callback

```php
function do_unsafe_thing(InputInterface $input, string $thing, callable $callback) {
    if ($input->getOption('dry-run')) {
        info("Not $thing because --dry-run option is set");
    } else {
        return $callback();
    }
}
```

and you'd call it like this:

```php
do_unsafe_thing($input, 'deleting label on GitHub', fn () => github_api($url, [], 'DELETE'));
```

I won't hold back merging if you decide not to do this, but it would tidy this up and give us a quick way to handle this for future standardisation scripts.

Reply:
Won't implement.

Given we'd need to pass a message and a callable, it'll just make the code more convoluted

Reply:
I disagree about making it more convoluted, but I'm okay with it not being implemented.